tune_sklearn/tune_search.py

```diff
@@ -314,10 +314,24 @@ def __init__(self,
 
         can_use_param_distributions = False
 
+        from ray.tune.schedulers import HyperBandForBOHB
         if search_optimization == "bohb":
             import ConfigSpace as CS
             can_use_param_distributions = isinstance(check_param_distributions,
                                                      CS.ConfigurationSpace)
+            if early_stopping is False:
+                raise ValueError(
+                    "early_stopping must not be False when using BOHB")
+            elif not isinstance(early_stopping, HyperBandForBOHB):
+                if early_stopping != "HyperBandForBOHB":
+                    warnings.warn(
+                        "Ignoring early_stopping value, as BOHB requires HyperBandForBOHB"
+                    )
+                early_stopping = "HyperBandForBOHB"
+        elif early_stopping == "HyperBandForBOHB" or isinstance(
+                early_stopping, HyperBandForBOHB):
+            raise ValueError(
+                "early_stopping can be HyperBandForBOHB only when using BOHB")
```

Comment:
`search_optimization must be set to 'BOHB' if early_stopping is set to HyperBandForBOHB.`